### ISSUE TITLE
Notebook execution seems slow as clicking Run All does not reach extension host for a few seconds

### ISSUE THREAD

**Issue by DonJayamanne** [microsoft/vscode#207755]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 
- OS Version: 

Steps to Reproduce:

1.  Create a large notebook with a few 100 cells (500 cells)
2.  Ensure the notebook has large outpus, 
Use the following to generate cell with outupt and copy this a few 100 times.
```python
from IPython import display
display.Image("/Users/donjayamanne/demo/output.png")
```
3. Open Jupyter output panel
4. Run one cell and wait for it to run
5. Clear Jupyter output panel
6. Hit run all
7. Notice that it takes a few seconds for the message `Handle Execution of Cells` to get logged in output panel.
On my machine it takes around 5s, meaning VS Code core is busy doing something before exension host gets the message to execute the cells.

This could be the same issue we're runnging into over SSH and other cases as well.
Basically users report Jupyter notebook is slow, but based on Jupyter extension logs, things are fast,
this seems to line up perfectly with the above.
